don't panic when random_kill is specified in local tests

## Bug Report

**What did you do?**

Use `LocalClusterProvisioner` as service provisoner, and then test a case with `random_kill` specified. 

**What did you expect to see?**
Although `random_kill` is not supported in local mode, it's better not to panic for tipocket.

**What did you see instead?**
However tipocket paniced:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x166c7f6]

goroutine 217 [running]:
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x204f4a0, 0xc00017e240, 0x0, 0x0, 0x20133e0, 0xc0007f8000, 0xc000899470, 0xc, 0xc0007b6090, 0xc0007b4040, ...)
        /home/qupeng/.go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/controller/controllerutil/controllerutil.go:142 +0x96
github.com/pingcap/tipocket/pkg/nemesis.(*Chaos).ApplyPodChaos(0xc0005fd950, 0x204f4a0, 0xc00017e240, 0xc0007f8000, 0x1d55e08, 0x0)
        /home/qupeng/.go/src/github.com/pingcap/tipocket/pkg/nemesis/client.go:55 +0xbe
github.com/pingcap/tipocket/pkg/nemesis.kill.Invoke(0xc0005fd950, 0x204f4a0, 0xc00017e240, 0xc0001f20a0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/qupeng/.go/src/github.com/pingcap/tipocket/pkg/nemesis/kill.go:153 +0x1ef
github.com/pingcap/tipocket/pkg/control.(*Controller).onNemesis(0xc00028a1e0, 0x204f4a0, 0xc00017e240, 0xc0007ae050)
        /home/qupeng/.go/src/github.com/pingcap/tipocket/pkg/control/control.go:459 +0x335
github.com/pingcap/tipocket/pkg/control.(*Controller).dispatchNemesis.func1(0xc0004b3f68, 0xc00028a1e0)
        /home/qupeng/.go/src/github.com/pingcap/tipocket/pkg/control/control.go:418 +0x45
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0007b6030, 0xc0007b6060)
        /home/qupeng/.go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/qupeng/.go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:54 +0x66
```

--- a/pkg/control/control.go
+++ b/pkg/control/control.go
@@ -455,6 +455,7 @@ func (c *Controller) onNemesis(ctx context.Context, op *core.NemesisOperation) {
 	nemesis := core.GetNemesis(string(op.Type))
 	if nemesis == nil {
 		log.Errorf("nemesis %s is not registered", op.Type)
+		time.Sleep(30 * time.Second)
 		return
 	}
 	log.Infof("run nemesis %s...", op.String())

--- a/pkg/nemesis/nemesis.go
+++ b/pkg/nemesis/nemesis.go
@@ -26,17 +26,20 @@ type k8sNemesisClient struct {
 }
 
 func init() {
-	client := k8sNemesisClient{New(tests.TestClient.Cli)}
-	core.RegisterNemesis(kill{client})
-	core.RegisterNemesis(podKill{client})
-	core.RegisterNemesis(containerKill{client})
-	core.RegisterNemesis(networkPartition{client})
-	core.RegisterNemesis(netem{client})
-	core.RegisterNemesis(scaling{client})
-	core.RegisterNemesis(timeChaos{client})
+	// most kinds of nemesis depends on chaos-mesh or tidb-operator
+	if tests.TestClient.Cli != nil {
+		client := k8sNemesisClient{New(tests.TestClient.Cli)}
+		core.RegisterNemesis(kill{client})
+		core.RegisterNemesis(podKill{client})
+		core.RegisterNemesis(containerKill{client})
+		core.RegisterNemesis(networkPartition{client})
+		core.RegisterNemesis(netem{client})
+		core.RegisterNemesis(timeChaos{client})
+		core.RegisterNemesis(iochaos{client})
+		core.RegisterNemesis(scaling{client})
+	}
 	core.RegisterNemesis(scheduler{})
 	core.RegisterNemesis(newLeaderShuffler())
-	core.RegisterNemesis(iochaos{client})
 }
 
 func shuffleIndices(n int) []int {

--- a/pkg/nemesis/pd_scheduler.go
+++ b/pkg/nemesis/pd_scheduler.go
@@ -47,8 +47,7 @@ func (s *schedulerGenerator) schedule(nodes []clusterTypes.Node, duration time.D
 	return ops
 }
 
-type scheduler struct {
-}
+type scheduler struct{}
 
 func (s scheduler) Invoke(ctx context.Context, node *clusterTypes.Node, args ...interface{}) error {
 	ctx, cancel := context.WithTimeout(context.TODO(), time.Minute*time.Duration(10))

--- a/tests/list_append/append.go
+++ b/tests/list_append/append.go
@@ -196,7 +196,6 @@ func (c *client) NextRequest() interface{} {
 }
 
 func (c *client) DumpState(_ context.Context) (interface{}, error) {
-	log.Printf("begin to create %d tables", c.tableCount)
 	for i := 0; i < c.tableCount; i++ {
 		if _, err := c.db.Exec(fmt.Sprintf("drop table if exists txn_%d", i)); err != nil {
 			return nil, err
@@ -213,7 +212,6 @@ func (c *client) DumpState(_ context.Context) (interface{}, error) {
 			}
 		}
 	}
-	log.Printf("create %d tables finished", c.tableCount)
 	return nil, nil
 }
 

